source/common/router/config_impl.h

```diff
@@ -389,7 +390,7 @@ class VirtualHostImpl : Logger::Loggable<Logger::Id::router> {
   std::shared_ptr<const SslRedirectRoute> ssl_redirect_route_;
   SslRequirements ssl_requirements_;
 
-  std::vector<RouteEntryImplBaseConstSharedPtr> routes_;
+  absl::InlinedVector<RouteEntryImplBaseConstSharedPtr, 2> routes_;
```

Comment:
The reserve lgtm, but if there is only one route, will the 2 here introduce more memory usage?